Applications/ApplicationsLib/ProjectData.cpp

```diff
@@ -352,3 +355,41 @@ void ProjectData::parseNonlinearSolvers(BaseLib::ConfigTree const& config)
 		    "The nonlinear solver name is not unique");
 	}
 }
+
+static std::unique_ptr<MathLib::PiecewiseLinearInterpolation>
+createPiecewiseLinearInterpolation(BaseLib::ConfigTree const& config)
+{
+	auto const coords = config.getConfParam<std::vector<double>>("coords");
+	auto const values = config.getConfParam<std::vector<double>>("values");
+	if (coords.empty() || values.empty())
+	{
+		ERR("The given co-ordinates or values vector is empty.");
+		std::abort();
+	}
+	if (coords.size() != values.size())
+	{
+		ERR("The given co-ordinates and values vector sizes are different.");
+		std::abort();
+	}
+
+	return std::unique_ptr<MathLib::PiecewiseLinearInterpolation>{
+	    new MathLib::PiecewiseLinearInterpolation{coords, values}};
```

Comment:
Fourth possibility is to implement ctor taking r-values only. Currently there are any calls with l-values.